apps/ukvi-complaints/lib/json-converters/brp.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+const Complaint = require('./complaint');
+
+class BrpComplaint extends Complaint {
+  constructor(values) {
+    super(values);
+    this.complaintAttributes.complaint.complaintType = 'BIOMETRIC_RESIDENCE_PERMIT';
+    this.complaintAttributes.complaint.complaintDetails.problemExperienced = this.problemExperiencedEnum();
+  }
+
+  problemExperiencedEnum() {
+    switch (this.values['biometric-residence-permit']) {
+      case 'card-incorrect':
+        return 'CARD_INCORRECT';
+      case 'card-not-arrived':
+        return 'CARD_NOT_ARRIVED';
+      case 'complain-brp':
+        return 'COMPLAIN_BRP';
+      default:
+        throw new Error('invalid "biometric-residence-permit" value');
+    }
```

Comment:
are all the switch/case statements necessary, could you just uppercase this.values and switch out hypens with underscores using gsub, rather than the need for all this?

credit to Alex Swann for that idea

Reply:
Annoyingly they aren't all exact matches in terms of wording. Did want to do that originally

Reply:
Alternatively you could have a module/file which is enums.js and is just an object of key/value pairs of enums which you import into where you need so like:

enums.js:
```
const enums = {
    'card-incorrect': 'CARD_INCORRECT',
    'card-not-arrived': 'CARD_NOT_ARRIVED',
    'complain-brp': 'COMPLAIN_BRP'
}
```

This is the same pattern used in translations - just here we are translating between two formattings rather than languages.

Then use like:
```
import enums from '/enums';

const myFormattedEnum = enums['card-incorrect']
```

Or something along those lines so in the end you have a single source of truth where all the enums for every complaint is stored. The dependancy can then be injected where used so when you test it it's easier to mock/stub.

Your Complaint class can then act like an abstract class which all your other complaint classes extend from, and on it have a `getFormattedEnum()` method like:

```
  getFormattedEnum(complaintType) {
    if (enums[complaintType]) {
      return enums[complaintType]
    }

    throw new Error(`invalid ${complaintType} value`);
  }
```

Doing it this way should mean you can remove all the getEnum style methods from all your complaint classes

Reply:
This was a good shout Rob. Have addressed this in the latest commit. Have tried to add all the ENUMs to a single file. Was a slight issue in that 'ihs' tag is used to mean two separate things with in the date returned by the form. I have accounted for this and it shouldn't be an issue.

Reply:
Have used this pattern. See commit 8bb2634